src/validation/validateProperties.ts

```diff
@@ -0,0 +1,173 @@
+                diagnostics.push(
+                    getPropertyNodeDiagnostic(prop, document, MESSAGES.INVALID_PROPERTY_NAME)
+                )
+            }
+        })
+
+        // if there is more than one item mark them all as invalid
+        if (itemsOfOneOfSet.length > 1) {
+            const messageSuffix = itemsOfOneOfSet
+                .map(item => item.propNode.keyNode.value)
+                .join(', ')
+
+            itemsOfOneOfSet.forEach(oneOfProp => {
+                diagnostics.push(
+                    getPropertyNodeDiagnostic(oneOfProp.propNode, document, MESSAGES.MUTUALLY_EXCLUSIVE_PROPERTIES + messageSuffix)
+                )
+            })
+        // if there is only one item and it is an object
+        // recursively continue on with validation
+        } else if (itemsOfOneOfSet.length) {
+            const { schemaValue, propNode } = itemsOfOneOfSet[0]
+            const { valueNode } = propNode
+            if (isObject(schemaValue)) {
+                diagnostics = diagnostics.concat(
+                    propertyAnalizer(valueNode, document, schemaValue, schemaSets)
+                )
+            }
+        }
+    // Fn:ValueOf
+    } else if (typeof valueOfSchema === 'string' && isObjectNode(rootNode)) {
+        const newSchemaPart: unknown = schemaSets[valueOfSchema]
+        diagnostics = diagnostics.concat(propertyAnalizer(rootNode, document, newSchemaPart, schemaSets))
+    // Regural properties
+    } else if (isObjectNode(rootNode)) {
+        rootNode.properties.forEach(prop => {
+            const propName = prop.keyNode.value
+
+            if (!schemaPart[propName]) {
+                diagnostics.push(
+                    getPropertyNodeDiagnostic(prop, document, MESSAGES.INVALID_PROPERTY_NAME)
+                )
+            }
+        })
+    }
+
+    return diagnostics
+}
+
+function getPropertyNodeDiagnostic(propNode: PropertyASTNode, document: TextDocument, message: string): Diagnostic {
+        const { length, offset } = propNode.keyNode
+        const range = Range.create(document.positionAt(offset), document.positionAt(offset + length))
+
+        return Diagnostic.create(range, message, DiagnosticSeverity.Error)
+}
+
+export default function(oneStateValueNode: ObjectASTNode, document: TextDocument): Diagnostic[] {
+    // Get the type of state
+    const typeNode = findPropChildByName(oneStateValueNode, 'Type')
+    const typeName = typeNode?.valueNode.value
+
+    let diagnostics: Diagnostic[] = []
+
+    oneStateValueNode.properties.forEach(prop => {
+        const propName = prop.keyNode.value
+        let hasCommonFields = true
+
+        // By default hasCommonFields should be true
+        // Only validate common fields when hasCommonFields is not false
+        if (typeof typeName === 'string') {
+            // tslint:disable-next-line no-unsafe-any
+            hasCommonFields = schema.Specific[typeName]?.hasCommonFields !== false
```

Comment:
I want it to be by default `true`. If it is not defined, it will be `undefined` and `!!` will turn it to `false`. Only when schema explicitly specifies it as `false` I want them to be disabled.

Reply:
That's true. If there's a type check on typeName it's already much more easy to read.